.gitmodules

```diff
@@ -1,3 +1,6 @@
 [submodule "coins/monero/c/monero"]
 	path = coins/monero/c/monero
 	url = https://github.com/monero-project/monero
+[submodule "coins/ethereum/schnorr-verify"]
+	path = coins/ethereum/schnorr-verify
+	url = https://github.com/noot/schnorr-verify.git
```

Comment:
Would you be open to inlining the contract from this and ensuring it has the needed tests within Rust, instead of a submodule? You would be able to maintain the copyright notice and GPL license structure :) (though I think AGPL would be better given how SCs are frequently considered backend...).

Reply:
can do, do you mean copy-pasting the entire repo basically? I'd like to eventually remove the need for JS altogether by compiling it within Rust which should be doable with ethers-solc

Reply:
Solely the ETH contract, which would require replacing the build sys as well. So not hardhat/JS.

Reply:
updated